src/systems/filecoin_mining/storage_proving/storage_proving_subsystem.go

```diff
@@ -64,17 +67,14 @@ func (sps *StorageProvingSubsystem_I) GenerateSurprisePoStCandidates(challengeSe
 
 	var poster = sps.PoStGenerator()
 
-	poster.GeneratePoStCandidates(challengeSeed, numChallengeTickets, sectorIDs)
-
-	todo := make([]sector.PoStCandidate, 0)
-	return todo
+	return poster.GeneratePoStCandidates(challengeSeed, numChallengeTickets, sectorIDs)
 }
 
-func (sps *StorageProvingSubsystem_I) CreateSurprisePoStProof(challengeSeed sector.PoStRandomness, challengeTickets []sector.PoStCandidate) sector.PoStProof {
+func (sps *StorageProvingSubsystem_I) CreateSurprisePoStProof(challengeSeed sector.PoStRandomness, candidates []sector.PoStCandidate) sector.PoStProof {
 	witness := &sector.PoStWitness_I{
-		Candidates_: challengeTickets,
+		Candidates_: candidates,
 	}
 
 	var poster = sps.PoStGenerator()
-	return poster.CreateSurprisePoStProof(witness)
+	return poster.CreateSurprisePoStProof(challengeSeed, witness)
```

Comment:
if you're able to review terminology in the `election_post.md` doc, "challengeSeed" is not used, "challengeTicket" is -- would be good to align. Not urgent.